Index: src/api/worker/search/EventQueue.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/api/worker/search/EventQueue.js	(revision 757d10fce26770569c808ac23b34709adbd4041f)
+++ src/api/worker/search/EventQueue.js	(date 1609942478986)
@@ -114,6 +114,7 @@
 
 	_optimizingAddEvents(newBatch: QueuedBatch, batchId: Id, groupId: Id, newEvents: $ReadOnlyArray<EntityUpdate>): void {
 		for (const newEvent of newEvents) {
+			this._optimizationEnabled && console.log("event in batch ", batchId, groupId, newEvent.type, newEvent.operation)
 			const elementId = newEvent.instanceId
 			const lastBatchForEntity = this._lastOperationForEntity.get(elementId)
 			if (lastBatchForEntity == null || this._processingBatch != null && this._processingBatch === lastBatchForEntity) {
@@ -187,10 +188,11 @@
 					this.removeEventsForInstance(elementId, firstMoveIndex + 1)
 
 				} else {
-					throw new ProgrammingError(`Impossible modification combination ${lastEntityModification} ${newEntityModification}`)
+					throw new ProgrammingError(`Impossible modification combination ${lastEntityModification} ${newEntityModification} ${JSON.stringify(newEvent)}`)
 				}
 			}
 		}
+		this._optimizationEnabled && console.log("end batch")
 	}
 
 
